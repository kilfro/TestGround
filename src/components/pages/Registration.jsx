import React from 'react';
import {connect} from 'react-redux';
import {Button, Container, TextField} from "@material-ui/core";
import '../../styles/component/pages/registration.css';
import {registerWithEmail} from "../../auth/auth";
import {createError} from "../../store/actions/error";
import {Redirect} from "react-router";
import logo from '../../logo.png';
import {loginSuccess} from "../../store/actions/auth";
import {createUser} from "../../api/api";

class Registration extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            successRegistration: false
        };
    }

    submitHandler = (event) => {
        event.preventDefault();
        const {createError, loginSuccess} = this.props;
        const email = document.getElementById('email').value,
            password = document.getElementById('password').value,
            repeatPassword = document.getElementById('repeat-password').value;
        if (password === repeatPassword) {
            registerWithEmail(email, password)
                .then(res => {
                    loginSuccess({uid: res.uid});
                    createUser({
                        uid: res.uid,
                        email
                    });
                })
                .catch(e => createError(e));
            this.setState({successRegistration: true});
        } else {
            createError('Пароли не совпадают')
        }
    };

    render() {
        const {authenticated} = this.props;
        const {successRegistration} = this.state;

        return (
            <Container maxWidth={'xs'} className={'registration-form-container'}>
                {(authenticated || successRegistration) && <Redirect to={'/'}/>}

                <img src={logo} alt={'logo'}/>
                <h3>Заполните форму для регистрации:</h3>
                <form onSubmit={this.submitHandler} className={'registration-form'}>
                    <TextField id={'email'} label={'E-mail'} type={'email'} fullWidth required/>
                    <TextField id={'password'} label={'Пароль'} type={'password'} fullWidth required/>
                    <TextField id={'repeat-password'} label={'Повторите пароль'} type={'password'} fullWidth required/>
                    <Button type={'submit'} color={'primary'} variant={'contained'}
                            fullWidth className={'ground_btn'}>Зарегистрироваться</Button>
                </form>
            </Container>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        authenticated: state.auth.authenticated
    }
};

const mapDispatchToProps = {
    createError,
    loginSuccess
};

export default connect(mapStateToProps, mapDispatchToProps)(Registration);