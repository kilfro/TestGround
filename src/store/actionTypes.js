export const AUTH = {
    LOGIN_WITH_GOOGLE: 'LOGIN_WITH_GOOGLE',
    LOGIN_WITH_EMAIL: 'LOGIN_WITH_EMAIL',
    LOGIN_SUCCESS: 'LOGIN_SUCCESS',
    LOGOUT_REQUEST: 'LOGOUT_REQUEST',
    LOGOUT_SUCCESS: 'LOGOUT_SUCCESS',
    REGISTER_REQUEST: 'REGISTER_REQUEST'
};

export const ERROR = {
    CREATE: 'CREATE_ERROR',
    DELETE: 'DELETE_ERROR'
};

export const NEW_TEST = {
    INSERT_TEST_TO_DB: 'NEW_TEST_INSERT_TEST',
    CHANGE_DESCRIPTION: 'NEW_TEST_CHANGE_DESCRIPTION',
    CREATE_UID: 'NEW_TEST_CREATE_UID',
    ADD_QUESTION: 'NEW_TEST_ADD_QUESTION',
    CHANGE_QUESTION: 'NEW_TEST_CHANGE_QUESTION',
    REMOVE_QUESTION: 'NEW_TEST_REMOVE_QUESTION',
    CLEAN_STATE: 'NEW_TEST_CLEAN_STATE',
    ADD_RESULT: 'NEW_TEST_ADD_RESULT',
    CHANGE_RESULT: 'NEW_TEST_CHANGE_RESULT',
    REMOVE_RESULT: 'NEW_TEST_REMOVE_RESULT'
};