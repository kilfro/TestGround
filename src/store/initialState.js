export default {
    error: null,
    user: {
        authenticated: false,
        photoURL: null,
        uid: null,
        displayName: null,
        token: null
    },
    newTest: {
        uid: '',
        description: {
            name: '',
            additional: '',
            password: '',
            anonymous: false,
            onlyRegistered: false,
            needPassword: false
        },
        questions: [
            {
                id: 1,
                multiple: false,
                cost: 1,
                question: '',
                options: [
                    {
                        id: 1,
                        text: '',
                        isRight: false
                    }
                ]
            }
        ],
        resultDescriptions: [
            {
                id: 1,
                min: 0,
                max: 0,
                text: ''
            }
        ]
    }
};