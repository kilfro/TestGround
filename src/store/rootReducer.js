import {combineReducers} from "redux";
import {authReducer} from "./reducers/auth";
import {errorReducer} from "./reducers/error";

export default combineReducers({
    auth: authReducer,
    error: errorReducer
});