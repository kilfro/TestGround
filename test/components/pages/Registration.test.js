import React from 'react';
import configureStore from 'redux-mock-store';
import {create} from 'react-test-renderer';
import {Provider} from "react-redux";
import Registration from "../../../src/components/pages/Registration";
import {ERROR} from "../../../src/store/actionTypes";

const mockStore = configureStore([]);

describe('Registration page', () => {
    let store, wrapper;
    const createError = jest.fn();

    beforeEach(() => {
        store = mockStore({
            auth: {
                authenticated: false
            }
        });
        store.dispatch = createError;

        wrapper = mount(
            <Provider store={store}>
                <Registration/>
            </Provider>
        );
    });

    afterEach(() => {
        wrapper.unmount();
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should create error', () => {
        wrapper.find('input#email').simulate('change', {target: {id: 'email', value: 'email@test.com'}});
        wrapper.find('input#password').simulate('change', {target: {id: 'password', value: 'password'}});
        wrapper.find('input#repeatPassword').simulate('change', {target: {id: 'repeatPassword', value: 'password2'}});
        wrapper.find('form').simulate('submit', {
            preventDefault: () => {
            }
        });

        expect(createError).toHaveBeenCalled();
        expect(store.dispatch).toHaveBeenCalledWith({payload: 'Пароли не совпадают', type: ERROR.CREATE});
    });
});