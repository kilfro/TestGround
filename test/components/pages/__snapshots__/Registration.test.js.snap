// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration page should render correctly 1`] = `
<div
  className="MuiContainer-root registration-form-container MuiContainer-maxWidthXs"
>
  <img
    alt="logo"
    src="test-file-stub"
  />
  <h3>
    Заполните форму для регистрации:
  </h3>
  <form
    className="registration-form"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
        htmlFor="email"
        id="email-label"
      >
        E-mail
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="email"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="email"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
        htmlFor="password"
        id="password-label"
      >
        Пароль
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="password"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
        htmlFor="repeat-password"
        id="repeat-password-label"
      >
        Повторите пароль
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="repeat-password"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained ground_btn MuiButton-containedPrimary MuiButton-fullWidth"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Зарегистрироваться
      </span>
    </button>
  </form>
</div>
`;
